kartothek/serialization/_generic.py

```diff
@@ -486,14 +486,26 @@ def filter_array_like(
             np.logical_and(array_like > value, mask, out=out)
         elif op == "in":
             value = np.asarray(value)
+            nullmask = pd.isnull(value)
+            if value.dtype.kind in ("U", "S", "O"):
+                unique_vals = np.unique(value[~nullmask])
+                value_ser = pd.Series(unique_vals, name="value")
+                arr_ser = pd.Series(array_like, name="array").to_frame()
+                matching_idx = (
```

Comment:
Please add a comment here to explain what you're doing

Reply:
what do you have in mind? "pandas join is faster than numpy.isin"?

Reply:
More like: left join user-input with row-group arrays, do something with nulls...

Reply:
And wrt the rationale, I think it'd be great if you can summarize this PR description and add it as a comment, so that anybody looking at the code can understand why for these dtypes we use different logic

Reply:
> More like: left join user-input with row-group arrays, do something with nulls...

My problem with stuff like this is that it essentially spells out what the code says. Array left join values is literally what's written there